### PR TITLE
feat: added implementation for getChunkStrategy() and added doc for max simultaneous downloads

### PR DESCRIPTION
**Pull request by zelfroster**
Added implementation for `getChunkStrategy()`.
Added JSON API doc for `getQueueSize()` and `setQueueSize()`.

### CODE REVIEW

File: src/ft/ftcontroller.cc [hunk 1]
```diff
@@ -1274,6 +1274,14 @@ bool ftController::setChunkStrategy(const RsFileHash& hash,FileChunksInfo::Chunk
 	mit->second->mCreator->setChunkStrategy(s) ;
 	return true ;
 }
+FileChunksInfo::ChunkStrategy ftController::getChunkStrategy(const RsFileHash& hash)
+{
+    std::map<RsFileHash,ftFileControl*>::iterator mit=mDownloads.find(hash);
+	if (mit!=mDownloads.end())
+	{
+	  return mit->second->mCreator->getChunkStrategy() ;
+	}
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftcontroller.cc:1283]: Is returning a boolean good, as the function is of integer type?

File: src/ft/ftcontroller.cc [hunk 2]
```diff
@@ -1274,6 +1274,14 @@ bool ftController::setChunkStrategy(const RsFileHash& hash,FileChunksInfo::Chunk
 	mit->second->mCreator->setChunkStrategy(s) ;
 	return true ;
 }
+FileChunksInfo::ChunkStrategy ftController::getChunkStrategy(const RsFileHash& hash)
+{
+    std::map<RsFileHash,ftFileControl*>::iterator mit=mDownloads.find(hash);
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftcontroller.cc:1279]: Alright, and please tell me what it does 😅

File: src/retroshare/rsfiles.h [hunk 3]
```diff
@@ -603,6 +603,14 @@ class RsFiles
 	        const RsFileHash& hash,
 	        FileChunksInfo::ChunkStrategy newStrategy ) = 0;
 
+	/**
+	 * @brief Get chunk strategy for the file getting downloaded
+	 * @jsonapi{development}
+	 * @param[in] hash file identifier
+	 * @return default chunk strategy for file
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/retroshare/rsfiles.h:610]: Ohh, sorry. Done 👍🏼.

File: src/ft/ftcontroller.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftcontroller.cc:1283]: I have changed it like this. Is it ok?
```
FileChunksInfo::ChunkStrategy ftController::getChunkStrategy(const RsFileHash& hash)
{
  RsStackMutex stack(ctrlMutex); /******* LOCKED ********/
  std::map<RsFileHash,ftFileControl*>::iterator mit=mDownloads.find(hash);
	if (mit!=mDownloads.end())
	{
	  return mit->second->mCreator->getChunkStrategy() ;
	}
  return false;
}
```

File: src/ft/ftcontroller.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (csoler) [reply src/ft/ftcontroller.cc:1279]: it protects the std::map from changing after the iterator mit has been found. If it does => you get a crash with 100% probability.

File: src/ft/ftcontroller.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (csoler) [reply src/ft/ftcontroller.cc:1283]: it's not what I meant: it's C++, not JS, so you cannot mixup types ;-) I mean: change the function prototype to that:

bool ftController::getChunkStrategy(const RsFileHash& hash, FileChunkInfo::ChunkStrategy& s);

[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftcontroller.cc:1283]: > it's not what I meant: it's C++, not JS, so you cannot mixup types ;-) I mean: change the function prototype to that:
> 
> bool ftController::getChunkStrategy(const RsFileHash& hash, FileChunkInfo::ChunkStrategy& s);

Ohh, so you wanted me to make it a boolean return type function.
Yeah, that's why I thought that this felt weird in a statically typed language.

File: src/ft/ftserver.cc [hunk 4]
```diff
@@ -437,6 +437,10 @@ bool ftServer::setChunkStrategy(const RsFileHash& hash,FileChunksInfo::ChunkStra
 {
 	return mFtController->setChunkStrategy(hash,s);
 }
+FileChunksInfo::ChunkStrategy ftServer::getChunkStrategy(const RsFileHash& hash)
```
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftserver.cc]: Done

File: src/ft/ftcontroller.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (zelfroster) [reply src/ft/ftcontroller.cc:1284]: Done

**Pull request merged**